core/src/client/types.rs

```diff
@@ -183,3 +183,16 @@ impl<T: Serialize> Serialize for Rich<T> {
         }
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TokenBalance {
+    pub contract_address: Address,
+    pub token_balance: String,
```

Comment:
I think both token_balance and error can be options,
error can remain a string I guess for now:)

i suggest token_balance to be a bytes32 (i think U256 is best suited
)